### PR TITLE
Font Library: add progress-bar while uploading font assets

### LINKED ISSUE #54399: Font Library: add progress-bar while uploading font assets
Should we add a progress bar while uploading font assets to indicate that to users visually?

### PR DESCRIPTION
**Pull request by hbhalodia**
## What?

- Add `ProgressBar` component while uploading the font assets.

## Why?

- Issue - Closes https://github.com/WordPress/gutenberg/issues/54399

## How?

- This PR adds the progress bar components and hide the drop zone and file upload while assets are being uploaded.

## Testing Instructions

1. Open site editor.
2. Go to styles and then typography option.
3. Select any style and there would be option for upload.
4. Add the font file there, you would see the progress bar for the moment till file uploads, and we get any notice.

### Testing Instructions for Keyboard

- None.

## Screenshots or screencast 
- Since the uploading is very fast, you would hardly see the progress bar in action.

[Font Library - Progerss Bar.webm](https://github.com/WordPress/gutenberg/assets/58802366/c2ea716e-e15a-47e3-8284-03e346781794)

### CODE REVIEW

**Review by mikachan — COMMENTED** [review]
Thanks for working on this, @hbhalodia!

Currently I'm seeing this in the `isUploading` state:

image

I think it would be good to centre align the progress bar and add some vertical spacing, similar to the mockup from @jasmussen [here](https://github.com/WordPress/gutenberg/issues/54399#issuecomment-1717212256).

**Comment by hbhalodia** [comment]
Sure, @mikachan - I would look into this asap.

Thanks.

**Comment by hbhalodia** [comment]
Hi @mikachan, I have revised the code and made the progress bar center aligned and added the vertical spacing. Also in addition to it, I had just showed the progress bar while uploading, since while upload it is good to show just a progress bar and nothing else. Let me know if we need the bottom text while uploading, I can revert that. 

Here is the screenshot how it would be while uploading.

Screenshot 2024-01-04 at 10 22 19 AM

Thank You!

File: packages/edit-site/src/components/global-styles/font-library-modal/local-fonts.js [hunk 1]
```diff
@@ -150,29 +156,39 @@ function LocalFonts() {
...
+						) }
+					/>
+				) }
+				{ isUploading && (
+					<FlexItem>
+						<Spacer margin={ 32 } />
+						<ProgressBar className="font-library-modal__upload-area__progress-bar" />
+						<Spacer margin={ 32 } />
+					</FlexItem>
+				) }
```
Comment (mikachan) [inline packages/edit-site/src/components/global-styles/font-library-modal/local-fonts.js]: I think we can achieve a more consistent layout by re-using the `font-library-modal__upload-area` class, as this already has a fixed height. Something like this:

```suggestion
				{ isUploading && (
					<FlexItem>
						<div className="font-library-modal__upload-area">
							<ProgressBar />
						</div>
					</FlexItem>
				) }
```

We don't want the progress bar to have a grey background, so we could adjust the CSS so the background colour is specific to the upload button:

```
button.font-library-modal__upload-area {
	background-color: #f0f0f0;

}
```

This should create a more consistent layout between the not-uploading is isUploading states:

| Not-uploading | isUploading |
| -------------- | ------------ |
| image | image |

**Review by mikachan — COMMENTED** [review]
Thanks @hbhalodia!

> Let me know if we need the bottom text while uploading, I can revert that.

After testing this out, I think we should leave this text in place, as it reduces the visual changes and layout shifts.

Maybe we should also move the Notice component below this text, to further reduce the visual changes, e.g.:

image


I've left an inline comment too, as I think we can re-use an existing CSS class rather than the Spacer components. Let me know what you think!

**Comment by hbhalodia** [comment]
>I think we should leave this text in place

Sure, Would be keeping as is.

>Maybe we should also move the Notice component below this text.

Yep, we can do it. Would update in the next commit.

Also, would address the inline comment that you have mentioned.

Thanks for the review!

File: packages/edit-site/src/components/global-styles/font-library-modal/local-fonts.js (hunk 1 above)
Comment (hbhalodia) [reply packages/edit-site/src/components/global-styles/font-library-modal/local-fonts.js]: Sure would be looking onto this.

Thanks.

**Comment by hbhalodia** [comment]
Hi @mikachan, have updated the PR with the required changes, Let me know if that works. Thanks.

| Not Uploading State | Uploading State |
| :-------------: | :-------------: |
| Screenshot 2024-01-04 at 4 59 26 PM | Screenshot 2024-01-04 at 5 02 20 PM |

**Review by mikachan — APPROVED** [review]
Looks great, thanks @hbhalodia! ✨

**Comment by hbhalodia** [comment]
Thanks, @mikachan 👍.

**Pull request merged**